code/headers/st7735_buffered.hpp

```diff
@@ -30,19 +41,38 @@ namespace r2d2::display {
          * @param y
          * @param data
          */
-        void set_pixel(uint16_t x, uint16_t y, const uint16_t data) override;
+        void set_pixel(uint16_t x, uint16_t y, const uint16_t data) override {
+
+            const uint16_t t =
+                __REV16(data); // make a copy and reverse byte order
+
+            // write pixel data to the buffer
+            this->buffer[x + (y * this->width)] = t;
```

Comment:
You could inline the expression and remove `t`

Reply:
i dont understand what you mean here

Reply:
```suggestion
            this->buffer[x + (y * this->width)] = __REV16(data);
```

Reply:
fixed here https://github.com/R2D2-2019/display/pull/36/commits/46cc67c102b3047c1f22c8ec9cf509b547babc86